lib/redlock/client.rb

```diff
@@ -160,31 +160,31 @@ def initialize(connection)
           if connection.respond_to?(:client)
             @redis = connection
           else
-            @redis = Redis.new(connection)
+            @redis = RedisClient.new(connection)
           end
           @redis.extend(ConnectionPoolLike)
         end
       end
 
       def lock(resource, val, ttl, allow_new_lock)
         recover_from_script_flush do
-          @redis.with { |conn| conn.evalsha Scripts::LOCK_SCRIPT_SHA, [resource], [val, ttl, allow_new_lock] }
+          @redis.call('EVALSHA', Scripts::LOCK_SCRIPT_SHA, 1, resource, val, ttl, allow_new_lock)
```

Comment:
Trying to debug why are locks aren't being found testing upgrade from 1.3.2 to 2.0.0

I think behavior here changed since the change to call doesn't return anything and therefore servers.select { lock } never returns a lock as found

Reply:
confirmed that since I'm passing in a redis client with a connection pool, it would work as `@redis.with { |conn| conn.call .... } ` but not `@redis.call`